File: hw1/Codes/gon_furkan.rs
// RUST
fn main() {
      use std::collections::HashMap;
      use std::collections::BTreeMap;
      println!("Initialization--------------");
    
      let mut mesnevii_nuriye = HashMap::new();
      
      mesnevii_nuriye.insert(String::from("Mukaddime"),  3 );
      mesnevii_nuriye.insert(String::from("Lem'alar"),  11 );
      mesnevii_nuriye.insert(String::from("Reshalar"),  12 );
      mesnevii_nuriye.insert(String::from("Lasiyyemalar"),  16 );
      mesnevii_nuriye.insert(String::from("Katre"),  33 );
      mesnevii_nuriye.insert(String::from("Hubab"),  34 );
      mesnevii_nuriye.insert(String::from("Habbe"),  32 );
      mesnevii_nuriye.insert(String::from("Zuhre"),  32 );
      mesnevii_nuriye.insert(String::from("Zerre"),  11 );
      mesnevii_nuriye.insert(String::from("Semme"),  13 );
      mesnevii_nuriye.insert(String::from("Onuncu Risale"), 25 );
      mesnevii_nuriye.insert(String::from("14. Resha") , 6 );
      mesnevii_nuriye.insert(String::from("Sule"),  9 );
      mesnevii_nuriye.insert(String::from("Nokta"),  14 );
      
      println! ("{:?}", &mesnevii_nuriye);  
      
    
      // Get the value of a key
      println!("Get the value of a key--------------");
      println!("{}", mesnevii_nuriye["Mukaddime"]);
      println!("{}", mesnevii_nuriye["Katre"]);
      println!("{}", mesnevii_nuriye["Hubab"]);
      println!("{}", mesnevii_nuriye["Nokta"]);
    
      // Add a new item 
      println!("Add a new item--------------");
      
      mesnevii_nuriye.insert(String::from("Fihrist") , 10);
      mesnevii_nuriye.insert(String::from("Itizar") , 1);
      
      println! ("{:?}", &mesnevii_nuriye);  
      
      // Remove an element
      println!("Remove an item--------------");
      
      mesnevii_nuriye.remove("Fihrist");
     
      println! ("{:?}", &mesnevii_nuriye);  
      
       // Modify an existing element
      println!("Modify an existing  item--------------");
      
      mesnevii_nuriye.insert(String::from("Itizar") , 3);
      
      println! ("{:?}", &mesnevii_nuriye);  
      
      // Existence of a key
      println!("Contains Key----------------");
      
      println!("{}", mesnevii_nuriye.contains_key("Habbe"));
      println!("{}", mesnevii_nuriye.contains_key("Sabah"));
      
      // Existence of a vaule
      println!("Contains Value----------------");
      
      {
          let v = 12;
          let does_contain = mesnevii_nuriye.values().any(|&val| val == v);
          println!("{:?}", does_contain);
      }
      
      {
          let v = 33;
          let does_contain = mesnevii_nuriye.values().any(|&val| val == v);
          println!("{:?}", does_contain);
      }
      
      {
          let v = 38;
          let does_contain = mesnevii_nuriye.values().any(|&val| val == v);
          println!("{:?}", does_contain);
      }
      
       // Function foo()--------------
      println!("Function foo----------------");
    
      fn foo(key: &String, value: &i32) {
          println!("mesneviiNuriye[{}] =  {}", key, value);
      }
  
      for (key, value) in &mesnevii_nuriye {
           foo(key, value);
      }
    
     
  }

File: hw1/Codes/gon_furkan.lua
--  // Initialization
print('Initialization------------')
 
mesneviiNuriye = {}

mesneviiNuriye["Mukaddime"] =  3
mesneviiNuriye["Lemalar"] =  11
mesneviiNuriye["Reshalar"] = 12
mesneviiNuriye["Lasiyyemalar"] = 16
mesneviiNuriye["Katre"] = 33
mesneviiNuriye["Hubab"] = 34
mesneviiNuriye["Habbe"] =  32
mesneviiNuriye["Zuhre"] =  32
mesneviiNuriye["Zerre"] =  11
mesneviiNuriye["Semme"] =  13
mesneviiNuriye["Onuncu Risale"] = 25
mesneviiNuriye["14. Resha"] = 6
mesneviiNuriye["Sule"] = 9
mesneviiNuriye["Nokta"] = 14

print(mesneviiNuriye)

--  // Get the value for a given key
print('Get the value of a key--------------')
print(mesneviiNuriye["Mukaddime"])
print(mesneviiNuriye["Katre"])
print(mesneviiNuriye.Hubab)
print(mesneviiNuriye.Nokta)

--  // Add a new item 
print('Add a new item--------------')

mesneviiNuriye["Fihrist"] = 10
mesneviiNuriye["Itizar"] = 1

print(mesneviiNuriye.Fihrist)
print(mesneviiNuriye.Itizar)

--  // Remove an element
print('Remove an item--------------')

mesneviiNuriye["Fihrist"] = nil
print(mesneviiNuriye["Fihrist"])


--   // Modify an existing element
print('Modify an existing  item--------------')

mesneviiNuriye["Itizar"] =  3

print(mesneviiNuriye["Itizar"])

--   // Existence of a key
print('Contains Key----------------')
print(mesneviiNuriye["Habbe"] ~= nil)
print(mesneviiNuriye["Sabah"] ~= nil)

--   // Existence of a vaule
print("Contains Value----------------")
--   
function contains(table, element)
    for _, value in pairs(table) do
        if value == element then
        return true
        end
    end
    return false
end

print(contains(mesneviiNuriye, 12))
print(contains(mesneviiNuriye, 33))
print(contains(mesneviiNuriye, 38))

--  // Function foo()--------------
print('Function foo----------------')

function foo(key, value)
    print(key, "=>", value)
end

for k,v in pairs(mesneviiNuriye) do
    foo(k, v)  
end
